cmd/cli/app/provider/provider_update.go

```diff
@@ -0,0 +1,509 @@
+}
+
+type parser func(string) (*reflect.Value, error)
+
+func parseBool(val string) (*reflect.Value, error) {
+	v, err := strconv.ParseBool(val)
+	if err != nil {
+		return nil, fmt.Errorf("expected bool, got %s", val)
+	}
+	res := reflect.ValueOf(v)
+	return &res, nil
+}
+
+func parseInt(size int) parser {
+	return func(val string) (*reflect.Value, error) {
+		v, err := strconv.ParseInt(val, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("expected integer, got %s", val)
+		}
+		res := reflect.ValueOf(v)
+		return &res, nil
+	}
+}
+
+func parseUint(size int) parser {
+	return func(val string) (*reflect.Value, error) {
+		v, err := strconv.ParseUint(val, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("expected integer, got %s", val)
+		}
+		res := reflect.ValueOf(v)
+		return &res, nil
+	}
+}
+
+func parseFloat(size int) parser {
+	return func(val string) (*reflect.Value, error) {
+		v, err := strconv.ParseFloat(val, size)
+		if err != nil {
+			return nil, fmt.Errorf("expected integer, got %s", val)
+		}
+		res := reflect.ValueOf(v)
+		return &res, nil
+	}
+}
+
+func identity(val string) (*reflect.Value, error) {
+	res := reflect.ValueOf(val)
+	return &res, nil
+}
+
+var (
+	parserMap = map[reflect.Kind]parser{
+		reflect.Bool:    parseBool,
+		reflect.Int32:   parseInt(32),
+		reflect.Int64:   parseInt(64),
+		reflect.Uint32:  parseUint(32),
+		reflect.Uint64:  parseUint(64),
+		reflect.Float32: parseFloat(32),
+		reflect.Float64: parseFloat(64), // double
+		reflect.String:  identity,       // identity
+	}
+)
+
+type configStruct struct {
+	*minderv1.ProviderConfig
+	//nolint:lll
+	GitHub *minderv1.GitHubProviderConfig `json:"github,omitempty" yaml:"github" mapstructure:"github" validate:"required"`
+	//nolint:lll
+	GitHubApp *minderv1.GitHubAppProviderConfig `json:"github_app,omitempty" yaml:"github_app" mapstructure:"github_app" validate:"required"`
+}
+
+// UpdateProviderCommand is the command for enrolling a provider
+//
+//nolint:gocyclo
+func UpdateProviderCommand(
+	ctx context.Context,
+	cmd *cobra.Command,
+	_ []string,
+	conn *grpc.ClientConn,
+) error {
+	// TODO: get rid of provider flag, only use class
+	providerName := viper.GetString("name")
+	if providerName == "" {
+		return cli.MessageAndError(
+			"missing mandatory flag",
+			errMissingProviderName,
+		)
+	}
+	project := viper.GetString("project")
+	if project == "" {
+		return cli.MessageAndError(
+			"missing mandatory flag",
+			errMissingProject,
+		)
+	}
+
+	fields := make(map[string]any)
+
+	setAttrs := viper.GetStringSlice("set-attribute")
+	unsetAttrs := viper.GetStringSlice("unset-attribute")
+
+	client := minderv1.NewProvidersServiceClient(conn)
+	resp, err := client.GetProvider(ctx, &minderv1.GetProviderRequest{
+		Context: &minderv1.Context{
+			Project: &project,
+		},
+		Name: providerName,
+	})
+	if err != nil {
+		return cli.MessageAndError("Failed to get provider", err)
+	}
+
+	provider := resp.GetProvider()
```

Comment:
do you think it makes sense to add a non-nil check for provider? I know it shouldn't happen if we check the error, so maybe it's not needed, but my C programmer past compels me to check all pointers :-)